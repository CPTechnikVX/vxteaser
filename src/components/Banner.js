import React          from 'react';
import PropTypes      from 'prop-types';
import ElementFactory from '../utils/ElementFactory';

export default class Banner extends React.PureComponent {

	constructor(props) {
		super(props);

		// bind
		this.onBannerClick = this.onBannerClick.bind(this);
	}

	onBannerClick = (event) => {
		event.preventDefault();
		event.stopPropagation();

		if (typeof this.props.onBannerClick === 'function') {
			this.props.onBannerClick();
		}
	}

	render() {
		const config          = this.props.config;
		const color           = this.props.config.color ? this.props.config.color : 'white';
		const backgroundColor = this.props.config.backgroundColor ? this.props.config.backgroundColor : '#262728';
		const windowWidth     = this.props.windowWidth;

		let BannerElement;
		let BannerContent;
		if (this.props.config.template === 'fixedHeight') {
			const fixedHeights = this.props.config.fixedHeights;

			BannerElement = ElementFactory.getBanner({color, backgroundColor, fixedHeights, windowWidth});
		}

		const childrenWithProps = React.Children.map(this.props.children,
			(child) => {
				return React.cloneElement(child, {
					windowWidth: windowWidth,
				});
			}
		);

		return (
			<BannerElement onClick={this.onBannerClick} className="teaser-banner">
				{childrenWithProps}
			</BannerElement>
		);

	}
}

Banner.propTypes = {
	config:        PropTypes.object,
	children:      PropTypes.node,
	onBannerClick: PropTypes.func,
	windowWidth:   PropTypes.number,
};

Banner.defaultProps = {
	config:      {},
	windowWidth: window.innerWidth,
};
