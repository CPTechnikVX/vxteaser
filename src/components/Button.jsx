import React from 'react';
import styled from 'styled-components';

class Banner extends React.PureComponent {
	render() {
		const Button = styled.a`    
			border-radius: 2px;
		    display: inline-block;
		    font-family: 'Roboto';
		    cursor: pointer;
		    min-width: 50px;
		    position: relative;
		    text-align: left;
		    text-decoration: none;
		    transition: all 0.15s linear;
		    vertical-align: top;
		    padding: 0.7em 5em 0.65em;
		    font-size: 1.7rem;
		    background-color: #1F93E9;
		    border: 1px solid #1F93E9;
		    color: #FFFFFF;
		    text-transform: none;
		`;

		return <Button onClick={this.props.onClick}>{this.props.children}</Button>;
	}
}

export default Banner;