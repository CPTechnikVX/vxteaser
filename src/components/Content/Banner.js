import React         from 'react';
import PropTypes     from 'prop-types';
import classnames    from 'classnames';
import Constants     from '../../utils/Constants';
import BannerContent from './BannerContent';
import ClickEvent    from '../../utils/ClickEvent';

/**
 * Main Banner component
 */
export default class Banner extends React.PureComponent {
	constructor(props) {
		super(props);

		// bind
		this.onClickFn = this.onClickFn.bind(this);
	}

	onClickFn(event) {
		event.preventDefault();
		event.stopPropagation();

		this.props.onClickFn(new ClickEvent(event, this, {
			link: this.props.link,
		}));
	}

	render() {
		const {children, config, modifier, theme, windowWidth} = this.props;

		if (config) {
			const classList = [];
			let videoUrl    = config.fixedHeights[0]['backgroundUrl'].endsWith('.mp4') ? config.fixedHeights[0]['backgroundUrl'] : '';
			const styleObj  = {
				backgroundImage: videoUrl ? "" : `url('${config.fixedHeights[0]['backgroundUrl']}')`, 
				height:          `${config.fixedHeights[1]['height']}px`,
			};
			
			if (windowWidth < config.fixedHeights[1]['greaterThan']) {
				videoUrl = config.fixedHeights[1]['backgroundUrl'].endsWith('.mp4') ? config.fixedHeights[1]['backgroundUrl'] : '';
				styleObj.backgroundImage = videoUrl ? "" : `url('${config.fixedHeights[1]['backgroundUrl']}')`;
				styleObj.height          = `${config.fixedHeights[0]['height']}px`;

				classList.push(Constants.ClassName.BannerNoSkew);
				classList.push('-view2');
			} else {
				classList.push(Constants.ClassName.Banner);
				classList.push('-view1');
			}

			if (theme) {
				classList.push('-theme-' + theme);
			}

			if (modifier) {
				classList.push(modifier);
			}

			return (
				<div className={classnames(classList)} style={styleObj} onClick={this.onClickFn} data-id={config.id}>
					{videoUrl && <video playsInline autoPlay muted loop style={{position: 'absolute', width: '100%', height: '100%', objectFit: 'cover'}}><source src={videoUrl} /></video>}
					<BannerContent config={config} windowWidth={windowWidth}>{children}</BannerContent>
				</div>
			);
		} else {
			return null;
		}
	}
}

Banner.propTypes = {
	/** @ignore */
	config:      PropTypes.object,
	/** @ignore */
	children:    PropTypes.node,
	modifier:    PropTypes.string,
	/** @ignore */
	onClickFn:   PropTypes.func,
	link:        PropTypes.string,
	theme:       PropTypes.string,
	/** @ignore */
	windowWidth: PropTypes.number,
};
