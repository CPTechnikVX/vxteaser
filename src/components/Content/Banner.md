```html
<banner theme="light" link="https://www.visit-x.net/de/magazin/aktuelle-ausgabe/">
    <skewpanelgroup skewWidth="54%">
      <panel width="45%" valign="bottom">
        <row>
          <col>
            <headline type="sub">Februar 2018</headline>
          </col>
        </row>
        <row>
          <col>
            <headline type="main">VISIT-X Magazin</headline>
          </col>
        </row>
        <row>
          <col>
            <headline>Exklusive Interviews mit den Camgirls, Live Einblicke in ein S/M Studio und die Chance das Traumauto von LexyRoxx zu gewinnen!</headline>
          </col>
        </row>
        <row>
          <col>
            <button type="primary">Jetzt lesen</button>
          </col>
        </row>
      </panel>
    </skewpanelgroup>
</banner>
```
```html
// banner has background position contain - for tablet (=> view2)
<banner modifier="-view2--bg-contain">
...
</banner>
```

```html
// banner with light theme
<banner modifier="-theme-light">
...
</banner>
```

```html
// banner with special theme
<banner modifier="-theme-special">
...
</banner>
```

```html
// banner with a transparent theme
<banner modifier="-theme-transparent">
...
</banner>
```
