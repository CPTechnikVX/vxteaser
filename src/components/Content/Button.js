import React      from 'react';
import PropTypes  from 'prop-types';
import Constants  from '../../utils/Constants';
import classnames from 'classnames';

/**
 * General button element
 */
export default class Button extends React.PureComponent {
	render() {
		const {children, color, link} = this.props;
		const classList               = [];
		let onClickFn;

		classList.push(Constants.ClassName.Button);

		if (color) {
			classList.push(Constants.PrefixClassName.Color + color);
		}

		if (link) {
			onClickFn = (e) => {
				e.preventDefault();
				e.stopPropagation();

				this.props.onLinkFn(link);
			};
		}

		return <a className={classnames(classList)} onClick={onClickFn}>{children}</a>;
	}
}

Button.propTypes = {
	/** @ignore */
	children: PropTypes.node,
	/**
	 * Textual color definition
	 */
	color:    PropTypes.oneOf(['primary']),
	link:     PropTypes.string,
	/** @ignore */
	onLinkFn: PropTypes.func,
};
