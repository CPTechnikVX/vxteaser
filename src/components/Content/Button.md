```js
<Button color="primary" link="https://www.example.com">Jetzt spielen!</Button>
```