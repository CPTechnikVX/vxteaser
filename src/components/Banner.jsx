import React from 'react';
import styled from 'styled-components';
import Button from './Button';

function Caption(props) {
	const CaptionElement = styled.div`    
			font-family: 'Roboto Condensed';
		    font-size: 2.6rem;
		    white-space: nowrap;
		`;

	return <CaptionElement>{props.config.text}</CaptionElement>;
}

function Headline(props) {
	const HeadlineElement = styled.div`
		    font-size: 4.5rem;
		    font-family: 'Roboto Condensed';
		    padding-bottom: 10px;
		    padding-top: 5px;
		    font-weight: bold;
		`;

	return <HeadlineElement>{props.config.text}</HeadlineElement>;
}

class Banner extends React.PureComponent {

	onBannerClick = (event) => {
		event.preventDefault();
		event.stopPropagation();

		alert('Banner clicked');
	};

	onButtonClick = (event) => {
		event.preventDefault();
		event.stopPropagation();

		alert('Teaser button clicked');
	};

	render() {
		const color           = 'white';
		const backgroundColor = '#262728';
		const backgroundUrl   = this.props.config.backgroundUrl ? this.props.config.backgroundUrl : '';

		let BannerElement;
		if (this.props.config.type === 'fixedHeight') {
			const fixedHeights = this.props.config.fixedHeights;

			BannerElement = styled.div`
				position:   relative;
				overflow:   hidden;
				width:      1300px;
				max-width:  100%;
				height:     ${fixedHeights[1]['height']}px;
				color:      ${color};
				background: ${backgroundColor} no-repeat url('${backgroundUrl}') top right;
				background-size: contain;
				
				&:before {
					position: absolute;
				    width: 55%;
				    height: 680px;
				    transform: rotate(26deg);
				    content: ' ';
				    top: -90%;
				    left: -9%;			
				    background-color: ${backgroundColor};
					@media (max-width: ${fixedHeights[1]['greaterThan']}px) {
		                display: none;
		            }
				}
				
				@media (max-width: ${fixedHeights[1]['greaterThan']}px) {
	                height: ${fixedHeights[0]['height']}px;
	            }
			`;
		}

		const BannerContent = styled.div`
			position: relative;
			padding: 2.5em 0 2.5em 2.5em;
			
			@media (max-width: 1200px) {
                margin-top: 260px;
                text-align: center;
            }
		`;

		const TextElement = styled.div`
		    font-size: 1.5rem;
		    line-height: 1.2;
		`;

		const ButtonContainer = styled.div`
			padding-top: 2em;
		`;

		return (
			<BannerElement onClick={this.onBannerClick} className="teaser-banner">
				<BannerContent>
					{this.props.children}
				</BannerContent>
				{/*<BannerContent>*/}
				{/*<Caption config={{text: this.props.config.title}}></Caption>*/}
				{/*<Headline config={{text: this.props.config.headline}}></Headline>*/}
				{/*<TextElement>*/}
				{/*Schau Dir hier unzensierte Bilder und Videos von der süßen 18 jährigen<br />*/}
				{/*an oder besuche sie bis zu 17 Stunden am Tag im Livechat.*/}
				{/*</TextElement>*/}
				{/*<ButtonContainer>*/}
				{/*<Button onClick={this.onButtonClick}>Zum Profil</Button>*/}
				{/*</ButtonContainer>*/}
				{/*</BannerContent>*/}
			</BannerElement>
		);

	}
}

Banner.defaultProps = {
	config: {},
};

export default Banner;