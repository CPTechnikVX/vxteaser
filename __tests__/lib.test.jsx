import React from 'react';
import renderer from 'react-test-renderer';
import Banner from '../src/components/Banner';
import Button from '../src/components/Button';

it('banner renders correctly', () => {
	const config = {
		title:         'SexyVany - Deutschlands j√ºngstes Camgirl',
		backgroundUrl: 'https://www.visit-x.net/assets/img/teaser/sexyvany/teaser-sexyvany.jpg',
	};

	const banner = renderer.create(
		<Banner config={config} />
	).toJSON();
	expect(banner).toMatchSnapshot();
});

it('button renders correctly', () => {
	const button = renderer.create(
		<Button>Text</Button>
	).toJSON();
	expect(button).toMatchSnapshot();
});
