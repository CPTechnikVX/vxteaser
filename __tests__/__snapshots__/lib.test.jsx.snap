// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`banner renders correctly 1`] = `
<div
  className="teaser-banner sc-bdVaJa gUCzSg"
  onClick={[Function]}
>
  <div
    className="sc-htpNat jAPvIx"
  >
    <div
      className="sc-bwzfXH gvnnog"
    >
      SexyVany - Deutschlands jüngstes Camgirl
    </div>
    <div
      className="sc-bxivhb gTyBKV"
    >
      EXKLUSIV AUF VISIT-X!
    </div>
    <div
      className="sc-ifAKCX gdGkiv"
    >
      Schau Dir hier unzensierte Bilder und Videos von der süßen 18 jährigen
      <br />
      an oder besuche sie bis zu 17 Stunden am Tag im Livechat.
    </div>
    <div
      className="sc-EHOje jDUGaz"
    >
      <a
        className="sc-bZQynM jtAGBJ"
      >
        Zum Profil
      </a>
    </div>
  </div>
</div>
`;

exports[`button renders correctly 1`] = `
<a
  className="sc-gzVnrw bZtgKD"
>
  Text
</a>
`;
